pkg/planner/core/point_get_plan.go

```diff
@@ -446,32 +386,25 @@ func (p *PointGetPlan) PrunePartitions(sctx sessionctx.Context) (bool, error) {
 	}
 	row := make([]types.Datum, len(p.TblInfo.Columns))
 	if p.HandleConstant == nil && len(p.IndexValues) > 0 {
-		partColsNames := pt.Meta().Partition.Columns
-		if len(partColsNames) > 0 {
-			partIdx, done, err := needsPartitionPruning(sctx, p.TblInfo, pt, p.dbName, p.IndexInfo, p.IdxCols, p.IndexValues, p.AccessConditions, p.PartitionNames)
-			if table.ErrNoPartitionForGivenValue.Equal(err) {
-				err = nil
-				partIdx = nil
-			}
-			if err != nil {
-				return false, err
-			}
-			if done {
-				if len(partIdx) == 1 {
-					p.PartitionIdx = &partIdx[0]
-					return false, nil
+		indexValues := p.IndexValues
+		evalCtx := sctx.GetExprCtx().GetEvalCtx()
+		for _, col := range p.IdxCols {
+			if !collate.IsBinCollation(col.GetType(evalCtx).GetCollate()) {
+				// If a non-binary collation is used, the values in `p.IndexValues` are sort keys and cannot be used for partition pruning.
+				r, err := ranger.DetachCondAndBuildRangeForPartition(sctx.GetRangerCtx(), p.AccessConditions, p.IdxCols, p.IdxColLens, sctx.GetSessionVars().RangeMaxSize)
```

Comment:
Why are we looping on `p.IdxCols` and doing this for the full conditions, and all columns (`p.IdxCols`)?
Should we also check if the column is part of the partitioning expression and then check if binary collation?

Reply:
Done